feat: update project cache

J1za opened the pull request:
## Description

## Linear Ticket
https://linear.app/autokitteh/issue/UI-855/cache-all-assets-on-first-load-files-triggers-connections-variables-to
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

<!--
     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.
     Before submitting a Pull Request, please ensure you've done the following:
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages (as described below).
     - 📗 Update any related documentation and include any relevant screenshots.
     Commit Message Structure (all lower-case):
     <type>(optional ticket number): <description>
     [optional body]
-->

RonenMars commented on src/components/organisms/deployments/table.tsx:
```diff
@@ -91,11 +92,11 @@ export const DeploymentsTable = () => {
 			{loadingDeployments && isInitialLoad ? <Loader isCenter size="xl" /> : null}
 
-			{!loadingDeployments && !deployments?.length ? (
+			{!deployments?.length ? (
```
```suggestion
			{!isInitialLoad && !deployments?.length ? (
```

To prevent flickering.
We don't want to display it before the first load.

J1za replied:
changed

RonenMars commented on src/components/organisms/topbar/project/projectConfig.tsx:
```diff
@@ -19,17 +19,27 @@ import {
-		if (!projectId || variant !== TopbaButtonVariant.actions) return;
+		if (!projectId) return;
+
+		initCache(projectId, true);
```
We don't need to init cache in case we're on sessions page.
We don't need an assets cache on the deployments page.
We don't need a deployments cache on the assets page.

J1za replied:
done

RonenMars commented:
<img src="https://media1.giphy.com/media/Hc8PMCBjo9BXa/giphy.gif"/>

RonenMars commented:
[test](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExNHZ3M2tyNzluMTJubm5nbG93YWMwcG44d3k3dTBkbHVhNXhzY2pkdCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/4GswqBy3owUQH4cLvp/giphy.gif)

RonenMars reviewed (commented):
1. Why is when we are going to the "Assets" tab - `fetchDeployments(projectId!, true);` triggered?
2. We don't need to `fetchEvents` on `initCache`, only on the events page.

J1za commented:
> 1. Why is when we are going to the "Assets" tab - `fetchDeployments(projectId!, true);` triggered?
> 2. We don't need to `fetchEvents` on `initCache`, only on the events page.

1. refactored
2. changed

RonenMars commented:
Creating a new project doesn't refresh the validation state:

https://github.com/user-attachments/assets/53e820b8-bf9f-4e5d-bc76-1e922eefbb78

RonenMars commented:
Moving between projects as well:

https://github.com/user-attachments/assets/d4f174bc-ade6-454d-a93e-ae3c4330bd51
